### PR TITLE
add query person by expanded fields

### PR DESCRIPTION
**Pull request by Kappaccinoh**
querypatient and querydoctor is able to now query by all other fields - now supports matching substring rather than whole words

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -38,6 +38,28 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+
+        String preppedSubstring = substring.trim();
+        checkArgument(!preppedSubstring.isEmpty(), "Substring parameter cannot be empty");
+
+        String preppedSentence = sentence.toLowerCase();
+        String preppedSubstringLower = preppedSubstring.toLowerCase();
+
+        return preppedSentence.contains(preppedSubstringLower);
+    }
+
```
Comment (alfaloo) [inline src/main/java/seedu/address/commons/util/StringUtil.java:62]: This feels leetcode style ahahaha

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 2]
```diff
@@ -123,6 +123,65 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
...
+    }
+
+    @Test
+    public void containsSubstringIgnoreCase_emptySubstring_throwsIllegalArgumentException() {
+        // Test empty substring
+        assertThrows(IllegalArgumentException.class, "Substring parameter cannot be empty", () -> {
+            StringUtil.containsSubstringIgnoreCase("example sentence", "");
+        });
+    }
+
```
Comment (alfaloo) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:184]: this looks nice

File: src/test/java/seedu/address/model/person/DoctorContainsKeywordsPredicateTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    @Test
+    public void toStringMethod() {
+        List<String> keywords = List.of("keyword1", "keyword2");
+        DoctorContainsKeywordsPredicate predicate = new DoctorContainsKeywordsPredicate(keywords);
+
+        String expected = DoctorContainsKeywordsPredicate.class.getCanonicalName()
+                + "{keywords=" + keywords + "}";
+        assertEquals(expected, predicate.toString());
+    }
+}
```
Comment (alfaloo) [inline src/test/java/seedu/address/model/person/DoctorContainsKeywordsPredicateTest.java:185]: I love these tests

File: src/test/java/seedu/address/model/person/PatientContainsKeywordsPredicateTest.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    @Test
+    public void toStringMethod() {
+        List<String> keywords = List.of("keyword1", "keyword2");
+        PatientContainsKeywordsPredicate predicate = new PatientContainsKeywordsPredicate(keywords);
+
+        String expected = PatientContainsKeywordsPredicate.class.getCanonicalName()
+                + "{keywords=" + keywords + "}";
+        assertEquals(expected, predicate.toString());
+    }
+}
```
Comment (alfaloo) [inline src/test/java/seedu/address/model/person/PatientContainsKeywordsPredicateTest.java:185]: damn thats a lot of tests

**Review by alfaloo — APPROVED** [review]

File: src/main/java/seedu/address/commons/util/StringUtil.java
Comment (Alteqa) [inline src/main/java/seedu/address/commons/util/StringUtil.java:1]: looks good

File: src/main/java/seedu/address/logic/parser/QueryDoctorCommandParser.java
Comment (Alteqa) [inline src/main/java/seedu/address/logic/parser/QueryDoctorCommandParser.java:1]: some minor changes looks ok

File: src/main/java/seedu/address/logic/parser/QueryPatientCommandParser.java
Comment (Alteqa) [inline src/main/java/seedu/address/logic/parser/QueryPatientCommandParser.java:1]: I feel like this is abit unecessary but it makes it cleaner I guess.

File: src/test/java/seedu/address/commons/util/StringUtilTest.java
Comment (Alteqa) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:1]: Damnnnn neat stuff boiii.

File: src/test/java/seedu/address/model/person/DoctorContainsKeywordsPredicateTest.java
Comment (Alteqa) [inline src/test/java/seedu/address/model/person/DoctorContainsKeywordsPredicateTest.java:1]: Ok nicely done.

File: src/test/java/seedu/address/model/person/PatientContainsKeywordsPredicateTest.java
Comment (Alteqa) [inline src/test/java/seedu/address/model/person/PatientContainsKeywordsPredicateTest.java:1]: Same with these.

**Review by Alteqa — COMMENTED** [review]
looks good

**Pull request merged**